### PR TITLE
[kube-prometheus-stack] Change thanos unused secret condition

### PR DESCRIPTION
**Pull request by bogdan-at-adobe**
#### What this PR does / why we need it

Following https://github.com/prometheus-community/helm-charts/pull/3918 a secret has been added for the thanos configs, however that secret gets applied regardless of whether the user uses it or not (if not used, it is just empty). This fix changes the condition to better match its purpose.

#### Which issue this PR fixes

It is not a direct related issue but the cause is the same in https://github.com/prometheus-community/helm-charts/issues/4033 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by Allex1 — APPROVED** [review]

File: charts/kube-prometheus-stack/templates/prometheus/secret.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.prometheus.enabled }}
+{{- if and .Values.prometheus.enabled .Values.prometheus.prometheusSpec.thanos.objectStorageConfig }}
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/templates/prometheus/secret.yaml:1]: What about

.Values.prometheus.prometheusSpec.thanos.objectStorageConfig.existingSecret

?

If defined, then this secret can be omit, too? Like line 13

Comment (bogdan-at-adobe) [reply charts/kube-prometheus-stack/templates/prometheus/secret.yaml]: I have applied the suggestion. Moved the if statement to the beginning of the manifest.

**Review by jkroepke — CHANGES_REQUESTED** [review]
LGTM 👍

**Review by jkroepke — DISMISSED** [review]
LGTM 👍

**Comment by bogdan-at-adobe** [comment]
The error seems to be on a grafana-test pod, I don't think this PR should interfere. Could you please do a retest? @jkroepke

**Review by jkroepke — APPROVED** [review]
LGTM 👍

**Pull request merged**